lua/telescope/actions/init.lua

```diff
@@ -420,6 +420,49 @@ actions.smart_send_to_qflist = function(prompt_bufnr)
   end
 end
 
+actions.complete_tag = function(prompt_bufnr)
+  local current_picker = action_state.get_current_picker(prompt_bufnr)
+  local tags = utils.get_default(current_picker.sorter.tags, {})
+  local delimiter = current_picker.sorter._delimiter
+
+  if vim.tbl_isempty(tags) then
+    print('No tags found or associated with picker.')
```

Comment:
I am not sure if we should not just exit silently when someone presses `<c-l>`, for example on accident in `find_files`.

Reply:
I initially followed your line of thought though then personally preferred verbosity plus possibly accounting for future scenarios I can't think of yet.

Nevertheless, your point of view is totally fair since, at least for the time being, there should not be scenarios in which `actions.complete_tag` can be intentionally called _and_ `tags` can be an empty table. An empty table should also eventually trigger `print(No matches found.)`.

So, let's settle for the below?

```lua
  local current_picker = action_state.get_current_picker(prompt_bufnr)
  local tags = current_picker.sorter.tags
  local delimiter = current_picker.sorter._delimiter

  -- silently exit for pickers that do not prefilter
  if not tags then return end
```

Reply:
Yeah i know i said hey lets give user feedback if he calls `Telescope lsp_*_diagnostics` but i am not sure how we should handle it here. Like someone hits it accidentally in `find_files` and just sees a random message about `tags`. 

But yeah i see what you saying, i also never know how to handle these cases

Reply:
No diagnostics for `lsp_*_diagnostics` is a much more common case though. Let's not confuse the 99% of telescope users still unaware of prefiltering with a half-hearted message as per the current commit they lack context to understand and be explicit instead

```lua
  local current_picker = action_state.get_current_picker(prompt_bufnr)
  local tags = current_picker.sorter.tags
  local delimiter = current_picker.sorter._delimiter

  if not tags then
    print('No tag pre-filtering set for this picker.')
    return
  end
```

since empty tables on second thought rightfully go to `print('No matches found.')`

If you'd agree we could additionally point to pickers for which tag pre-filtering is set out-of-the-box as a gimmick to increase discoverability of the feature like
`print('No tag pre-filtering set for this picker. Try, for instance, <C-l> for lsp_*_diagnostics or lsp_*_symbols!)` or something similar, since accidentally hitting `<C-l>` is also not that common I suppose, but the above is fine of course, too.

Reply:
Yeah works now, thanks :)
I was also thinking maybe we should just map that action for functions that actually support pre-filtering.

I am going to test it now :)